pkg/registry/generic/etcd/etcd.go

```diff
@@ -403,6 +403,7 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 			if lastGraceful > 0 {
 				return out, nil
 			}
+			return out, nil
 			// fall through and delete immediately
```

Comment:
I mean, you have an if block that does the same thing as your normal return.  So you need the case, but you should remove the if block.